### PR TITLE
Relax version dep of the "contracts" gem and require ruby v2.3+

### PR DESCRIPTION
**Pull request by mcls**
There's no breaking changes in newer versions of that gem, so I don't see a reason to set the dependency as tightly as it was set before. 

_(See the [`contracts` changelog](https://github.com/egonSchiele/contracts.ruby/blob/master/CHANGELOG.markdown#v0160---2017-04-24) for reference.)_

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://github.com/snowplow/snowplow/wiki/CLA to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

**Comment by mcls** [comment]
ℹ️  I signed it ^

**Comment by snowplowcla** [comment]
@mcls  has signed the Software Grant and Corporate Contributor License Agreement

File: snowplow-tracker.gemspec [hunk 1]
```diff
@@ -34,7 +34,7 @@ Gem::Specification.new do |s|
   s.platform    = Gem::Platform::RUBY
   s.required_ruby_version = '>= 2.0.0'
```
Comment (BenFradet) [inline snowplow-tracker.gemspec]: should we change this since we seem to support only >= 2.3?

**Comment by mscwilson** [comment]
Hi @mcls, massive apologies for taking so long to get back to you.

I've cherry-picked your `contracts` version commit for release v0.7.0. However, we'd like to continue supporting older versions of Ruby (>= 2.1) for now. 

Thank you for your contributions that are helping to improve the Ruby tracker. I see that you have another PR open, which will also be incorporated into the next release.

**Pull request closed without merging**